### ISSUE TITLE
Free positioning of items in grid

### ISSUE THREAD

**Issue by klehmann** [haltu/muuri#453]
Hi Niklas! Your project looks really cool! We are searching for a grid solution where users can position the items wherever they want in the grid, e.g. some items top left, others bottom right etc.

So no autolayout requirement in that mode, just D&D to a free space with snap to nearest grid position.

Do you still this can be built? Is that possible out of the box or do we need a custom layout?

**Comment by niklasramo** [comment]
Yes, it can be built with Muuri, but you will need to create a custom layout algorithm and probably custom drag sort predicate.

**Comment by voyti** [comment]
Hey, is there any documentation on how to proceed with this (I've seen some custom layout implementations, but nothing close or touching a custom drag sort predicate)? I'm looking into staying with Muuri in our project, but this requirement seems to be a bit complicated to start with without some more detailed documentation

**Comment by stefanocurnis** [comment]
> Hi Niklas! Your project looks really cool! We are searching for a grid solution where users can position the items wherever they want in the grid, e.g. some items top left, others bottom right etc.
> 
> So no autolayout requirement in that mode, just D&D to a free space with snap to nearest grid position.
> 
> Do you think this can be built? Is that possible out of the box or do we need a custom layout?
> 
> To make this work, we would need to get the drop x/y position so that we can compute the nearest grid position.

this is interesting. did you solve this? I'm about to write custom stuff to achieve this

**Comment by klehmann** [comment]
No, not yet. There are examples of http://gridstackjs.com that have this kind of layout. So I will probably go that route if that requirement come up again in the future.
